Increase the deadline since we are using less memory

kevinkirkup opened the pull request:
Also use the default number of jobs, which should equal the number
of available cores.

dmittendorf commented on pureport/resource_aws_connection.go:
```diff
@@ -0,0 +1,313 @@
+			Optional: true,
+			Elem:     &schema.Schema{Type: schema.TypeString},
+		},
+		"peering": {
```
We should name this `peering_type` for consistency with the Ansible module https://github.com/pureport/pureport-ansible-modules/blob/develop/ansible/modules/pureport/pureport_aws_direct_connect_connection.py

kevinkirkup replied:
ok

dmittendorf commented on pureport/resource_azure_connection.go:
```diff
@@ -0,0 +1,285 @@
+			Required: true,
+			ForceNew: true,
+		},
+		"peering": {
```
Please rename to `peering_type` for consistency with Ansible modules.

kevinkirkup replied:
ok

dmittendorf commented on pureport/resource_google_cloud_connection.go:
```diff
@@ -0,0 +1,282 @@
+	}
+
+	// Google Optionals
+	if secondaryPairingKey, ok := d.GetOk("secondary_pairing_key"); ok {
```
Is there a way to require `secondaryPairingKey` when highAvailability is true?

kevinkirkup replied:
Let me check.

kevinkirkup replied:
Doesn't look like there is, but it should be validated by the server so we should be ok.

dmittendorf commented on pureport/resource_network.go:
```diff
@@ -0,0 +1,239 @@
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"account_id": {
```
`account_href`?

kevinkirkup replied:
Same as above. The Account ID is required by the API.

```
func (a *NetworksApiService) AddNetwork(ctx context.Context, accountId string, localVarOptionals *AddNetworkOpts) (*http.Response, error)
```

kevinkirkup commented on pureport/connection.go:
```diff
@@ -0,0 +1,343 @@
+			Required: true,
+			ForceNew: true,
+		},
+		"network": {
```
Currently for networks, we need both the `id` and the `href` so kept it as a link object.

kevinkirkup commented on pureport/connection.go:
```diff
@@ -0,0 +1,343 @@
+
+			new := client.CustomerNetwork{
+				Name:    cn["name"],
+				Address: cn["Address"],
```
yes

kevinkirkup commented on README.md:
```diff
@@ -1,2 +1,89 @@
+Maintainers
+-----------
+
+This provider plugin is maintained by the Terraform team at [HashiCorp](https://www.hashicorp.com/).
```
Yeah, this was just a cut/paste from one of the existing Terraform roles.

kevinkirkup commented on pureport/data_source_locations.go:
```diff
@@ -0,0 +1,149 @@
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"links": {
```
This is actually the `LocationLinkConnection` not the `Link` type from Kato, so I think we are OK here.
Agree on the remove of Link from this interface though.

kevinkirkup commented on pureport/data_source_cloud_services.go:
```diff
@@ -0,0 +1,141 @@
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+						"cloud_region_id": {
```
Currently in the API, this is referencing the `ID` of the cloud region that the service is located, not the `href` so it this case, I think it's OK.

i.e. 
1	*aws-s3-us-west-1*	AWS S3 us-west-1	AWS	S3	4	FALSE		1557431944

kevinkirkup replied:
We can revisit this. We may be able to get rid of this completely.
Not sure that it's required.

kevinkirkup replied:
Or abstract, the interface to only expose the provider id.
Need to think about that a bit more.

kevinkirkup replied:
I'm thinking in general what we need to do here is provide both the `id` and `href` parameters for reading but only require `href` for writing `Link` objects.
There are a couple of reasons for this:
* There are a few of places in the golang SDK that require providing an ID (`accountID` for creating networks, and `networkID` for creating connections.
* It's easier to centralize the HREF parser for extracting the ID in a single location. i.e. Kato and have just be able to provide this to clients.

Thoughts?

kevinkirkup replied:
Moving to use `HREF` everywhere.

kevinkirkup commented on pureport/data_source_networks.go:
```diff
@@ -0,0 +1,134 @@
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"account_id": {
```
Same here. The API is referencing the `ID`

ie. 
```
func (a *NetworksApiService) FindNetworks(ctx context.Context, accountId string) ([]Network, *http.Response, error)
```

Try75 commented on go.mod:
```diff
@@ -0,0 +1,13 @@
+	github.com/antihax/optional v0.0.0-20180407024304-ca021399b1a6
+	github.com/cenkalti/backoff v2.1.1+incompatible
+	github.com/hashicorp/terraform v0.11.13
+	github.com/pureport/pureport-sdk-go v0.0.0-20190506215644-d100251c07ec
```
Does this need to be updated to a released version of pureport-sdk-go?

kevinkirkup replied:
Probably.

kevinkirkup replied:
Should be fixed with the latest PR.

Try75 commented on version/version.go:
```diff
@@ -0,0 +1,6 @@
+
+var (
+	// ProviderVersion is set during the release process to the release version of the binary
+	ProviderVersion = "dev"
```
Might need to change this for release, too.

kevinkirkup replied:
So it looks like I took this from the Google Terraform Module and it looks like they just leave this as `dev`. I think this is to just define the `ProviderVersion` variable and default it to `dev`. During their build process, they specify the version on the build command line, which is what we will probably do as well.

Commit:
Increase the deadline since we are using less memory

Also use the default number of jobs, which should equal the number
of available cores.
